src/wallet/mineTransaction.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { EventEmitter } from 'events';
+import { MIN_POLLING_INTERVAL } from '../constants';
+import Transaction from '../models/transaction';
+import txMiningApi from '../api/txMining';
+
+// Error to be shown in case of no miners connected
+const noMinersError = 'There are no miners to resolve the proof of work of this transaction.';
+
+// Error to be shown in case of an unexpected error
+const unexpectedError = 'An unexpected error happened. Please try to send your transaction again.';
+
+// Error to be shown in case of a timeout
+const timeoutError = 'Timeout solving transaction\'s proof-of-work.\n\nAll transactions need to solve a proof-of-work as an anti spam mechanism. Currently, Hathor Labs provides this service for free, but their servers may be fully loaded right now.';
+
+type promiseReturnType = {
+  nonce: number,
+  weight: number,
+  timestamp: number,
+  parents: string[],
+};
+
+/**
+ * This is transaction mining class responsible for:
+ *
+ * - Submit a job to be mined;
+ * - Update mining time estimation from time to time;
+ * - Get back mining response;
+ *
+ * It emits the following events:
+ * 'job-submitted': after job was submitted;
+ * 'estimation-updated': after getting the job status;
+ * 'job-done': after job is finished;
+ * 'error': if an error happens;
+ * 'unexpected-error': if an unexpected error happens;
+ **/
+class MineTransaction extends EventEmitter {
```

Comment:
Duplicate code?

Reply:
The idea was to migrate the `SendTransaction` class to typescript, then I adapted the code a bit for the new wallet service requirements.

Reply:
Can't we make one method call the other and remove duplication?

Reply:
The idea is that the old class will be deleted eventually.

Reply:
Done here: https://github.com/HathorNetwork/hathor-wallet-lib/pull/184/commits/552aa00899aab7cc45b80bfac8254bcdbac7c335

Reply:
Can you see these two commits? 
https://github.com/HathorNetwork/hathor-wallet-lib/pull/184/commits/8b2f4cd5ffdbccfc5924834ba4f3590cc1ba96dd
https://github.com/HathorNetwork/hathor-wallet-lib/pull/184/commits/faa4e9c82833580f95218a1aa9d7d85bedbeb2b1

The first is the refactor to remove duplicate code and the second is because the first one was not working for create token tx.

Reply:
Awesome! Thanks!